sdk/component/internal/storer/local/sqlite/sqlite.go

```diff
@@ -3,42 +3,69 @@ package sqlite
 import (
 	"context"
 	"database/sql"
+	"encoding/json"
+	"errors"
 	"fmt"
 	"path"
+	"time"
 
+	"github.com/jonboulle/clockwork"
 	_ "github.com/mattn/go-sqlite3"
+	"google.golang.org/protobuf/encoding/protojson"
 
+	"github.com/smithy-security/smithy/sdk/component/internal/storer"
+	"github.com/smithy-security/smithy/sdk/component/internal/uuid"
 	ocsf "github.com/smithy-security/smithy/sdk/gen/com/github/ocsf/ocsf_schema/v1"
 )
 
-const errInvalidConstructorEmptyReason = "cannot be empty"
+const (
+	errInvalidConstructorEmptyReason = "cannot be empty"
+
+	columnNameFindings   columnName = "findings"
+	columnNameWorkflowID columnName = "workflow_id"
+	columnNameUpdatedAt  columnName = "updated_at"
+)
 
 type (
 	manager struct {
+		clock  clockwork.Clock
 		db     *sql.DB
 		dbName string
 	}
 
+	managerOption func(*manager) error
+
+	columnName string
+
 	// ErrInvalidConstructor should be used for invalid manager constructor errors.
 	ErrInvalidConstructor struct {
 		argName string
 		reason  string
 	}
 )
 
+// ManagerWithClock allows customising manager's clock.
+func ManagerWithClock(clock clockwork.Clock) managerOption {
+	return func(m *manager) error {
+		if clock == nil {
```

Comment:
I think this check should be outside of the function so that it fails immediately instead of failing when the function is invoked.

Reply:
The one outside can't return an error as it returns an option - commonly done in the functional options pattern :)